contracts/proxy/TransparentUpgradeableProxy.sol

```diff
@@ -67,7 +67,7 @@ contract TransparentUpgradeableProxy is UpgradeableProxy {
      * https://eth.wiki/json-rpc/API#eth_getstorageat[`eth_getStorageAt`] RPC call.
      * `0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103`
      */
-    function admin() external ifAdmin returns (address admin_) {
+    function admin() external virtual ifAdmin returns (address admin_) {
```

Comment:
This and `implementation` are meant to be used as `view` functions, but Solidity doesn't allow to mark them as `view` because of the `ifAdmin` modifier. But given that they are intended as `view` I think I would remove the `virtual` modifier.